invenio_rdm_records/resources/iiif.py

```diff
@@ -48,6 +50,24 @@
 )
 
 
+def abort_record_deleted_exception(func):
+    """Abort deleted record exception.
+
+    Service methods may raise the Exception RecordDeletedException to show that
+    the method was called for a deleted record. The exception should be hidden
+    from the user.
+    """
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except RecordDeletedException:
+            abort(404)
```

Comment:
discussion: should we use another error code here?

Reply:
IMHO we should be consistent and display the same code (410) for all the record related endpoints:
![Screenshot from 2024-07-10 09-06-06](https://github.com/inveniosoftware/invenio-rdm-records/assets/15194802/2d154a01-b7ff-423c-bbdc-e6aa15257db9)

WDYT?

Reply:
i was not aware of this error code. is it enough to do `abort(410)` or do i have to implement something that the return dict will look like the json in the screenshot?

Reply:
Hmm, I think this is a bit more complex, the best/easiest solution I can think of is to simply add the error_handler that catches that error (RecordDeletedException) in the iiif resource layer returning a 410 with a message stating that the record is not available anymore, here: https://github.com/inveniosoftware/invenio-rdm-records/blob/6f028b22c5c2ea91a3df9162f9e385f3b5b5439c/invenio_rdm_records/resources/iiif.py#L94-L100

By doing this we are not displayig the full information about why the record was deleted but this is stored in the record data, maybe it wouldn't be too difficult to pass that data, but to me it would be enoughto have the 410 and a message saying the record was deleted. WDYT?

Reply:
something like https://github.com/inveniosoftware/invenio-rdm-records/pull/1758/commits/3c655e7f357c8910e743cbb52f846138de013c82 ?